src/twisted/internet/_resolver.py

```diff
@@ -199,25 +239,25 @@ def resolveHostName(
         """
         # If it's str, we need to make sure that it's just ASCII.
         try:
-            hostName = hostName.encode("ascii")
+            hostName_bytes = hostName.encode("ascii")
         except UnicodeEncodeError:
             # If it's not just ASCII, IDNA it. We don't want to give a Unicode
             # string with non-ASCII in it to Python 3, as if anyone passes that
             # to a Python 3 stdlib function, it will probably use the wrong
             # IDNA version and break absolutely everything
-            hostName = _idnaBytes(hostName)
+            hostName_bytes = _idnaBytes(hostName)
 
         # Make sure it's passed down as a native str, to maintain the interface
-        hostName = nativeString(hostName)
+        hostName = nativeString(hostName_bytes)
 
         resolution = HostResolution(hostName)
         resolutionReceiver.resolutionBegan(resolution)
         onAddress = self._simpleResolver.getHostByName(hostName)
```

Comment:
This line now doesn't match the interface for `IResolverSimple` which states that the `timeout` parameter is required. I'm not sure if I should:

1. Add a default tiemout here.
2. Update the interface to reflect that a default value exists.

Any thoughts on what to do here?

(There's a similar error related to `IHostnameResolver.resolveHostName` on line 348.)

Reply:
We chatted a bit about this in #twisted:

```
12:04:11 PM - clokep: Not sure how to get feedback on an ongoing PR, but I left a question at https://github.com/twisted/twisted/pull/11589#discussion_r961057512 -- the tl;dr is Twisted internally seems to assume that implementations of an interface have a default value which the interface doesn't show. What's the proper solution to this?
12:04:37 PM - clokep: (To add the default to the interface or the provide the defaults in the caller or something else?)
12:07:49 PM - jean-paul[m]: The goal should usually be to fix the mismatch without breaking any code.  Sometimes that's hard, maybe even hard enough that it's not worth doing, but that's the goal that should guide choices.
12:08:35 PM - jean-paul[m]: Since it's basically a point of extension, we should expect developers are implementing the interface as currently defined.
12:09:48 PM - jean-paul[m]: Buuuut if they did that, their implementation wouldn't actually work with Twisted, right?  Because Twisted calls the method without a value for that parameter.
12:10:28 PM - jean-paul[m]: So from that perspective, the interface is wrong and should be updated to have a default - reflecting real usage by Twisted.
12:11:26 PM - jean-paul[m]: You could also look at it from the opposite perspective - someone wrote code that expects implementations of the interface and then uses it according to what the interface says.  Fortunately, this means those implementations would already be passing a value for `timeout` since the interface says you have to.
12:11:40 PM - jean-paul[m]: So this code also wouldn't be broken by adding a default to the interface.
12:12:08 PM - jean-paul[m]: I don't think that's a complete case analysis but it's probably the major points.  And I don't think this interface is an extremely popular extension point anyway.
12:12:21 PM - clokep: Right, my inclination was that adding a default to the interface is the "least" breaking thing to do.
12:12:40 PM - jean-paul[m]: That seems right to me.
12:13:01 PM - clokep: Awesome. Thanks for the help! Will try to finish that up "soon"!
12:13:23 PM - clokep: (I think connectTCP has a similar issue with parameters, will need to check the Synapse codebase again for ignores.)
12:15:00 PM - graingert[m]: I think the policy on type annotations is that they may break at any time right now
12:15:16 PM - graingert[m]: I can't find the docs on that
12:15:17 PM - graingert[m]: But they're currently provisional
12:15:42 PM - clokep: Yeah, I'm not really talking about the type annotations breaking -- I'm talking about type annotations finding bugs in old code. 😁
12:17:13 PM - graingert[m]: Ah right
```

The tl;dr is that option 2 from above seems more reasonable in this case (to avoid breaking code).